packages/fast-tooling-react/src/navigation/navigation.utilities.ts

```diff
@@ -758,38 +758,32 @@ export function getDataWithDuplicate<T>(sourceDataLocation: string, data: T): T
     if (isInArray(clonedData, normalizedSourceDataLocation)) {
         duplicateDataInArray(clonedData, normalizedSourceDataLocation);
     } else {
-        // check for case of single child as object
-        const sourceDataLocationSegments: string[] = sourceDataLocation.split(".");
-        if (
-            sourceDataLocationSegments[sourceDataLocationSegments.length - 2] ===
-            childrenKeyword
-        ) {
-            duplicateDataInArray(clonedData, normalizedSourceDataLocation);
-        }
+        duplicateData(clonedData, normalizedSourceDataLocation);
     }
 
     return clonedData;
 }
 
+function duplicateData(data: unknown, sourceDataLocation: string): void {
+    const sourceData: unknown = get(data as object, sourceDataLocation);
+    const targetData: unknown = [sourceData, sourceData];
```

Comment:
this isn't really unknown - its an array of unknowns

Reply:
That’s true